src/core.ts

```diff
@@ -364,6 +364,12 @@ export type TrackingConsentDetails = {
   updated?: boolean;
   /** Whether or not the UI has been shown to the end-user (undefined in older versions of airgap.js) */
   prompted?: boolean;
+  /** Transparency Consent (TCF) String */
+  tcf?: string;
+  /** US Privacy (USP) String */
+  usp?: string;
+  /** Global Privacy Platform (GPP) String */
+  gpp?: string;
```

Comment:
If we do it this way, purposes is always required, though IAB strings are not. This is fine for Airgap (since Airgap will always have purposes) but not for the backend (which may need to be able to get and set TCF strings without Airgap purposes).

Reply:
Ahh I forgot about the possibility of strings without airgap purposes. I guess, I can update purposes to be optional here and on the backend.

Reply:
@linh-transcend it should be optional for both purposes and the other consent strings, but we should also make sure to validate that at least one of those fields are defined in the app code

Reply:
could we include the view state and airgap version that was shown when consent was collected? - new requirements in canadian law

Reply:
FYI @MadDataScience I can make a task to add viewState and airgap version.

Reply:
it can be a follow up if needed, but just wanted to flag here in case its easier to do some of the work along the way

Reply:
Hm. Well, we can add it to the codec now. Might be easier to do that right away, so we don't have to go back and forth again when we update the monorepo.